backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/database/entity/SpecificationEntity.kt

```diff
@@ -3,23 +3,29 @@ package com.tngtech.apicenter.backend.connector.database.entity
 import org.hibernate.search.annotations.Field
 import org.hibernate.search.annotations.Indexed
 import org.hibernate.search.annotations.IndexedEmbedded
-import java.util.UUID
-import javax.persistence.CascadeType
-import javax.persistence.Column
-import javax.persistence.Entity
-import javax.persistence.Id
-import javax.persistence.OneToMany
-import javax.persistence.OrderBy
+import javax.persistence.*
 
 @Entity
 @Indexed
-data class SpecificationEntity(
-    @Id val id: UUID,
+class SpecificationEntity(
+    @Id val id: String,
     @Field val title: String,
     @Field @Column(columnDefinition = "TEXT") val description: String?,
     @IndexedEmbedded @OneToMany(
         mappedBy = "specification",
         cascade = [CascadeType.ALL]
     ) @OrderBy("created DESC") val versions: List<VersionEntity>,
     val remoteAddress: String?
-)
\ No newline at end of file
+) {
+
+    fun pureUpdate(other: SpecificationEntity): SpecificationEntity {
```

Comment:
1. Correct, in my mind the SpecificationEntity is only used like a transfer object passed to the repository, it doesn't really "store" data in a persistence sense. So it seemed misleading to have a method modify the object directly, which is why I made it "pure" / "immutable" rather than "in-place". </br></br>2. `other` is intended as the "newer" spec, the one which overrides the title / description with more recent content. Because the `id` is not supposed to be changed, it is preserved </br></br>3. Not sure. I assume the Version ordering is determined by queries to the repository in the `@OrderBy` annotation? I don't know if the ordering of the `VersionEntity`s here takes effect.